### PR TITLE
fit more info into success screen

### PR DESCRIPTION
**Pull request by TychoVrahe**
Introduces more sophisticated success animation to fit more info.

Some refactoring was done to frame: mostly i extracted header to separate component to not complicate the frame code with animations. As a side effect, this also safe some flash as the header does not need to be generic. I also extracted the animation from frame to save some more, overall these changes save about 7kB of flash. Some of it is claimed by the animations so net savings about 4kB.

Edit: added one more commit with frame code reshuffling, saving additional ~7kB.  https://github.com/trezor/trezor-firmware/pull/4023/commits/4c23fb8b99fded1c5071cc2d9aa7c0b3f9780936

### CODE REVIEW

File: core/translations/en.json [hunk 1]
```diff
@@ -947,6 +947,8 @@
     "words__unknown": "Unknown",
     "words__warning": "Warning",
     "words__writable": "Writable",
-    "words__yes": "Yes"
+    "words__yes": "Yes",
+    "result__title_success": "Success",
+    "result__title_neutral": "Done"
```
Comment (mmilata) [inline core/translations/en.json:952]: Nit: consider putting these under `words__title_`

**Review by mmilata — APPROVED** [review]
beautiful

File: core/translations/en.json (hunk 1 above)
Comment (TychoVrahe) [reply core/translations/en.json]: changed https://github.com/trezor/trezor-firmware/commit/14029121467f59c97ce2584c79d88c88570d72d3

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 24.7.3 (19fd64f5b2ab1bc265821e34d4c4b9c910864d0e)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.7.3 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.1 regular (revision 008490bf4b8a04a085c59e217f743c9713727480)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**